R/impact_central.R

```diff
@@ -299,3 +307,254 @@ get_birth_cohort <- function(data) {
   }
   data$year - data$age
 }
+
+
+#' Calculate impact by calendar year
+#'
+#' Calculate impact accrued over all ages for a specific year. This calculates
+#' the difference in disease burden between baseline and focal scenarios for a
+#' given year. The baseline scenario can have no vaccination or different
+#' coverage to the focal scenario. This aggregates the impact over all ages
+#' modelled. This does not account for the future disease burden averted through
+#' current vaccine activities.
+#'
+#' @param baseline_impact Data frame of baseline impact data this needs to have
+#' columns country, burden_outcome, year, age, value
+#' @param focal_impact Data frame of focal impact data this needs to have
+#' columns country, burden_outcome, year, age, value
+#'
+#' @return Vaccine impact by country and year for burden outcomes as a data
+#' frame with columns country, year, burden_outcome and impact
+#' @export
+impact_by_calendar_year <- function(baseline_impact, focal_impact) {
+  assert_has_columns(baseline_impact,
```

Comment:
I wonder if semantic validation functions here might be better (i.e., validate that these are sensible impact/baseline inputs), especially as the column checks are duplicated directly below, then again in the following function. Would we need to and anything to check that they are compatible (e.g., same countries/years/etc)?